test/e2e-helm/e2e_helm_suite_test.go

```diff
@@ -55,33 +57,16 @@ var _ = BeforeSuite(func() {
 	By("preparing the prerequisites on cluster")
 	tc.InstallPrerequisites()
 
-	By("initializing a Helm project")
-	err = tc.Init(
-		"--plugins", "helm",
-		"--project-version", "3-alpha",
-		"--domain", tc.Domain)
-	Expect(err).NotTo(HaveOccurred())
-
-	By("using dev image for scorecard-test")
-	err = tc.ReplaceScorecardImagesForDev()
-	Expect(err).NotTo(HaveOccurred())
-
-	By("creating an API definition")
-	err = tc.CreateAPI(
-		"--group", tc.Group,
-		"--version", tc.Version,
-		"--kind", tc.Kind)
-	Expect(err).NotTo(HaveOccurred())
+	By("mocking Helm Memcached Sample")
+	ctx, err := samples.NewSampleContextWithTestContext(&tc)
+	Expect(err).Should(Succeed())
+	sample := helm.NewMemcachedHelm(&ctx)
+	sample.Run()
```

Comment:
Why are we regenerating the sample in the e2e test? Let's just use the already generated sample in `testdata`.

Reply:
IMO: It would not be a good approach:
- We need to do changes in the e2e tests on top of the samples. For Ansible, for example, the e2e test mock needs to be Memcached-operator + molecule
- the e2e tests create its context (TestContext) which has other data which is not part of the SampleContext. 
- it would increase the complexity to work with and contribute by the project since after doing a change and run the e2e tests we would need also to re-gen all samples again to came back it for its state.  Otherwise, it will not pass in the test-sanity. 

Anyway, let's do it for all and then, after we make all work we can see what are the best halls for improvement. WDYT could we agree with?

Reply:
> We need to do changes in the e2e tests on top of the samples. For Ansible, for example, the e2e test mock needs to be Memcached-operator + molecule

Taking a step back, theoretically the samples checked into `testdata` should be immediately testable and they should pass all tests, so I would expect very few necessary modifications (basically just modifications that would be required to run binaries and base images built from the PR, e.g. `:dev` image replacements)

> the e2e tests create its context (TestContext) which has other data which is not part of the SampleContext.

I don't quite follow this. It seems like an implementation detail we can change to support running our normal e2e suite against existing files.

> it would increase the complexity to work with and contribute by the project since after doing a change and run the e2e tests we would need also to re-gen all samples again to came back it for its state. Otherwise, it will not pass in the test-sanity.

We can avoid that by copying the entire sample into a temporary directory at the beginning of the test and then deleting that directory at the end of the test. The actual `testdata` directory does not need to be touched.

Reply:
Hi @joelanford, 

I am not against new idea and improvements at all. I am all in for that. However, I'd like to move forward here with what was proposed in operator-framework/enhancements#47 and do NOT change the current scope/scenarios used in the e2e. 

See that it is not part of the scope of the EP we reduce the scope of the E2E tests. if we do what you are suggesting we will reduce by sure the scope of the e2e tests used for Ansible. We cannot perform for example the Ansible e2e with the memcached-operator sample mock only as you suggested. The mock used to these tests so far are mainly memcached-operator + the molecule steps. 

in this way, my idea here is 
- we implement the EP operator-framework/enhancements#47
- then, after we will have the same e2e tests/checks been done with the samples we will be in a better position to do cleanup and/or improvements as you suggested. 

Could we agree in re-check these options after all requirements are addressed and we have a better view over how we can improve it? Also, note that will be easier to check if something broke or not with the suggestions made after that.